cmd/consul-dataplane/config_test.go

```diff
@@ -103,9 +103,189 @@ func TestConfigGeneration(t *testing.T) {
 			wantErr: false,
 		},
 		{
-			desc: "able to override all the config fields with CLI flags",
+			desc: "able to generate config properly with proxy flags when the config file input is empty",
 			flagOpts: func() (*FlagOpts, error) {
-				opts, err := generateFlagOpts()
+				return generateFlagOptsWithProxyFlags()
+			},
+			makeExpectedCfg: func(flagOpts *FlagOpts) *consuldp.Config {
+				return &consuldp.Config{
+					Consul: &consuldp.ConsulConfig{
+						Addresses:           stringVal(flagOpts.dataplaneConfig.Consul.Addresses),
+						GRPCPort:            intVal(flagOpts.dataplaneConfig.Consul.GRPCPort),
+						ServerWatchDisabled: true,
+						Credentials: &consuldp.CredentialsConfig{
+							Type: "static",
+							Static: consuldp.StaticCredentialsConfig{
+								Token: "test-token-123",
+							},
+							Login: consuldp.LoginCredentialsConfig{
+								AuthMethod:  "test-iam-auth",
+								BearerToken: "bearer-login",
+							},
+						},
+						TLS: &consuldp.TLSConfig{
+							Disabled:           false,
+							CACertsPath:        "/consul/",
+							CertFile:           "ca-cert.pem",
+							KeyFile:            "key.pem",
+							ServerName:         "tls-server-name",
+							InsecureSkipVerify: true,
+						},
+					},
+					Proxy: &consuldp.ProxyConfig{
+						NodeName:  "test-node-dc1",
+						NodeID:    "dc1.node.id",
+						Namespace: "default",
+						ProxyID:   "node1.service1",
+						Partition: "default",
+					},
+					Logging: &consuldp.LoggingConfig{
+						Name:     DefaultLogName,
+						LogJSON:  true,
+						LogLevel: "WARN",
+					},
+					DNSServer: &consuldp.DNSServerConfig{
+						BindAddr: "127.0.0.1",
+						Port:     8604,
+					},
+					XDSServer: &consuldp.XDSServer{
+						BindAddress: "127.0.1.0",
+						BindPort:    0,
+					},
+					Envoy: &consuldp.EnvoyConfig{
+						AdminBindAddress:              "127.0.1.0",
+						AdminBindPort:                 18000,
+						ReadyBindAddress:              "127.0.1.0",
+						ReadyBindPort:                 18003,
+						EnvoyConcurrency:              4,
+						EnvoyDrainStrategy:            "test-strategy",
+						ShutdownDrainListenersEnabled: true,
+						GracefulShutdownPath:          "/graceful_shutdown",
+						EnvoyDrainTimeSeconds:         30,
+						GracefulPort:                  20300,
+					},
+					Telemetry: &consuldp.TelemetryConfig{
+						UseCentralConfig: true,
+						Prometheus: consuldp.PrometheusTelemetryConfig{
+							RetentionTime: 10 * time.Second,
+							ScrapePath:    "/metrics",
+							MergePort:     12000,
+							CACertsPath:   "/consul/",
+							CertFile:      "prom-ca-cert.pem",
+							KeyFile:       "prom-key.pem",
+						},
+					},
+				}
+			},
+			wantErr: false,
+		},
+		{
+			desc: "able to override all the config fields with CLI flags when using service flags",
```

Comment:
IMHO this test case is less interesting to duplicate because none of the flags overlap with service vs proxy config. However, `"test whether CLI flag values override the file values` does have a "service" config block, so I do think there should be a test there.

Reply:
I think this test is still valuable. I believe it's testing that every possible config value can be set via CLI flags. In this case, we need to still be able to check that if I provide proxy flags, those end up in dataplane config (even without overlapping service flags). We could create a test where we're only testing that subset of flags but I thought given that service flags are deprecated, it'd be easier to remove in the future because we can just keep the other test case and remove this one without sacrificing any test coverage.

>"test whether CLI flag values override the file values does have a "service" config block, so I do think there should be a test there.

I've added two more cases. One to test that cli flags override file values when using proxy flags and another one to test that proxy flags are preferred over service flags.